### PR TITLE
Cluster command fixed for public-uri parameter

### PR DESCRIPTION
**Pull request by ojalatodd**
Added --public-uri argument to cluster command.
Fixed for both subprocess and docker.

Signed-off-by: Todd Ojala <todd@bitwise.io>

### CODE REVIEW

File: manage/sawtooth_manage/docker.py [hunk 1]
```diff
@@ -135,10 +135,10 @@ def start(self, node_config):
         if node_config.genesis:
             command = 'bash -c "sawtooth admin keygen && \
             sawtooth admin genesis && \
-            validator {} -v"'
+            validator {} -v --public-uri tcp://0.0.0.0:8800"'
```
Comment (jsmitchell) [inline manage/sawtooth_manage/docker.py]: 0.0.0.0 is not valid for --public-uri. The intent is this is a reachable endpoint - it should be a hostname (preferably) or an ip address for how this docker container is reachable by other peers on the cluster.

Comment (ojalatodd) [inline manage/sawtooth_manage/docker.py]: @jsmitchell Fixed, I believe.

Comment (ojalatodd) [inline manage/sawtooth_manage/docker.py]: In subprocess, all the validators are running locally on the same machine, so subprocess should be OK, but take a look.

File: manage/sawtooth_manage/subproc.py [hunk 2]
```diff
@@ -103,13 +103,14 @@ def start(self, node_args):
             if cmd == 'validator':
                 component = '--component-endpoint', url
                 network = '--network-endpoint', gossip_port
+                public_uri = '--public-uri', gossip_port
                 peer_list = ['tcp://0.0.0.0:' + str(base_gossip_port + i)
```
Comment (jsmitchell) [inline manage/sawtooth_manage/subproc.py]: These are not valid peer addresses.

File: manage/sawtooth_manage/subproc.py [hunk 3]
```diff
@@ -103,13 +103,14 @@ def start(self, node_args):
             if cmd == 'validator':
                 component = '--component-endpoint', url
                 network = '--network-endpoint', gossip_port
+                public_uri = '--public-uri', gossip_port
```
Comment (jsmitchell) [inline manage/sawtooth_manage/subproc.py]: gossip_port, above is not a valid public-uri reachable endpoint

Comment (jsmitchell) [inline manage/sawtooth_manage/subproc.py]: If the purpose of this is a number of processes running on the same machine, then --public-uri should probably be something like "tcp://localhost:{}".format(gossip_port_num)

Comment (ojalatodd) [inline manage/sawtooth_manage/subproc.py]: Changed using your idea.

File: manage/sawtooth_manage/subproc.py (hunk 2 above)
Comment (ojalatodd) [inline manage/sawtooth_manage/subproc.py]: Noted. Subprocess was originally coded with 0.0.0.0 as the network address, but I have changed it to "localhost," after consultation with Boyd and others.

**Comment by ojalatodd** [comment]
Fixed merge conflict.

**Comment by ojalatodd** [comment]
@jsmitchell Conflict resolved, changes made.

File: manage/sawtooth_manage/docker.py [hunk 4]
```diff
@@ -135,14 +135,14 @@ def start(self, node_config):
...
             sawtooth admin genesis && \
-            validator {} -v"'
+            validator {} -v --public-uri tcp://{}:8800"'
         else:
             command = 'bash -c "sawtooth admin keygen && \
-            validator {} -v"'
+            validator {} -v --public-uri tcp://{}:8800"'
         if len(peers) > 0:
-            command = command.format('--peers ' + ",".join(peers))
+            command = command.format('--peers ' + " ".join(peers), node_name)
```
Comment (jsmitchell) [inline manage/sawtooth_manage/docker.py]: This needs to be a ",".join based on the new peers behavior

Comment (ojalatodd) [inline manage/sawtooth_manage/docker.py]: Fixed this for both docker and subprocess.

**Comment by ojalatodd** [comment]
Note: tests pass locally, but not on the build server.

**Pull request merged**